mockgen/generator.go

```diff
@@ -0,0 +1,385 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/token"
+	"log"
+	"path"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/golang/mock/mockgen/model"
+	toolsimports "golang.org/x/tools/imports"
+)
+
+type generator struct {
```

Comment:
It would be cool if the generator could be exported. This would allow to vendor it with the project, so code generation can happen offline.